cli/tests/services.bats

```diff
@@ -1307,3 +1320,89 @@ EOF
   run [ ! -e "$PWD/overmind.sock" ]
   assert_success
 }
+
+@test "activate: picks up changes after environment modification when all services have stopped" {
+
+  export FLOX_FEATURES_SERVICES=true
+
+  MANIFEST_CONTENTS_1="$(cat << "EOF"
+    version = 1
+
+    [services]
+    one.command = "echo $FOO"
+
+    [hook]
+    on-activate = "export FOO=foo_one"
+EOF
+  )"
+
+  "$FLOX_BIN" init
+  echo "$MANIFEST_CONTENTS_1" | "$FLOX_BIN" edit -f -
+
+  # Edit the manifest adding a second service and changing the value of FOO.
+  # Then start services again.
+  "$FLOX_BIN" activate -s -- sleep infinity &
+  activate_pid="$!"
+
+  # Make sure we avoid a race of service one failing to complete
+  for i in {1..5}; do
+    if "$FLOX_BIN" services status | grep "Completed"; then
```

Comment:
(non-blocking) I added a helper in https://github.com/flox/flox/pull/1932 that we can refactor these tests to use later.